crates/vm-superio/src/serial.rs

```diff
@@ -914,4 +1013,110 @@ mod tests {
         // THR empty interrupt was raised nevertheless.
         assert_eq!(iir & IIR_THR_EMPTY_BIT, IIR_THR_EMPTY_BIT);
     }
+
+    #[test]
+    fn test_serial_state_constructor() {
+        let state = SerialState {
+            baud_divisor_low: DEFAULT_BAUD_DIVISOR_LOW,
+            baud_divisor_high: DEFAULT_BAUD_DIVISOR_HIGH,
+            interrupt_enable: DEFAULT_INTERRUPT_ENABLE,
+            interrupt_identification: DEFAULT_INTERRUPT_IDENTIFICATION,
+            line_control: DEFAULT_LINE_CONTROL,
+            line_status: DEFAULT_LINE_STATUS,
+            modem_control: DEFAULT_MODEM_CONTROL,
+            modem_status: DEFAULT_MODEM_STATUS,
+            scratch: DEFAULT_SCRATCH,
+            in_buffer: Vec::new(),
+        };
+
+        assert_eq!(state.baud_divisor_low, DEFAULT_BAUD_DIVISOR_LOW);
+        assert_eq!(state.baud_divisor_high, DEFAULT_BAUD_DIVISOR_HIGH);
+        assert_eq!(state.interrupt_enable, DEFAULT_INTERRUPT_ENABLE);
+        assert_eq!(
+            state.interrupt_identification,
+            DEFAULT_INTERRUPT_IDENTIFICATION
+        );
+        assert_eq!(state.line_control, DEFAULT_LINE_CONTROL);
+        assert_eq!(state.line_status, DEFAULT_LINE_STATUS);
+        assert_eq!(state.modem_control, DEFAULT_MODEM_CONTROL);
+        assert_eq!(state.modem_status, DEFAULT_MODEM_STATUS);
+        assert_eq!(state.scratch, DEFAULT_SCRATCH);
+        assert!(state.in_buffer.is_empty());
+    }
+
+    #[test]
+    fn test_serial_state_default() {
+        let state = SerialState::default();
+
+        assert_eq!(state.baud_divisor_low, DEFAULT_BAUD_DIVISOR_LOW);
+        assert_eq!(state.baud_divisor_high, DEFAULT_BAUD_DIVISOR_HIGH);
+        assert_eq!(state.interrupt_enable, DEFAULT_INTERRUPT_ENABLE);
+        assert_eq!(
+            state.interrupt_identification,
+            DEFAULT_INTERRUPT_IDENTIFICATION
+        );
+        assert_eq!(state.line_control, DEFAULT_LINE_CONTROL);
+        assert_eq!(state.line_status, DEFAULT_LINE_STATUS);
+        assert_eq!(state.modem_control, DEFAULT_MODEM_CONTROL);
+        assert_eq!(state.modem_status, DEFAULT_MODEM_STATUS);
+        assert_eq!(state.scratch, DEFAULT_SCRATCH);
+        assert!(state.in_buffer.is_empty());
+    }
+
+    #[test]
+    fn test_from_state_with_too_many_bytes() {
```

Comment:
Another test case that would make sense considering our design choices of excluding implementation details from the state is the following:
- save the state of the serial console (let's name this S1) when there are bytes in the FIFO
- restore the serial from the previously saved state (let's name this S2)
- make sure that the S1 and S2 FIFOs are the same; what is important here is to check the order of the bytes

Reply:
Doesn't 
```
166     #[test]
167     fn test_state_ser() {
```
do exactly this?

Reply:
It's close enough to what I was suggesting, we can mark this as resolved.